### PR TITLE
Update dependencies from dotnet/sdk

### PR DESCRIPTION
**Pull request by dsplaisted**
Includes changes from #10765, plus updated stage 0 and other changes to make tests pass again

### CODE REVIEW

File: TestAssets/DesktopTestProjects/AppWithProjTool2Fx/App.csproj [hunk 1]
```diff
@@ -6,7 +6,7 @@
     <OutputType>Exe</OutputType>
     <AssetTargetFallback Condition=" '$(TargetFramework)' == 'netcoreapp3.0' ">$(AssetTargetFallback);portable-net45+win8;dnxcore50</AssetTargetFallback>
     <RuntimeIdentifiers>win7-x64;win7-x86;osx.10.10-x64;osx.10.11-x64;ubuntu.14.04-x64;ubuntu.16.04-x64;ubuntu.16.10-x64;centos.7-x64;rhel.7.2-x64;debian.8-x64;fedora.24-x64;opensuse.42.1-x64</RuntimeIdentifiers>
-    <RestoreAdditionalProjectSources Condition="'$(TEST_PACKAGES)' != ''">$(TEST_PACKAGES)</RestoreAdditionalProjectSources>
+    <RestoreAdditionalProjectSources Condition="'$(TEST_PACKAGES)' != ''">$(RestoreAdditionalProjectSources);$(TEST_PACKAGES)</RestoreAdditionalProjectSources>
```
Comment (nguerrera) [inline TestAssets/DesktopTestProjects/AppWithProjTool2Fx/App.csproj]: Why did this only become an issue now?

Comment (dsplaisted) [reply TestAssets/DesktopTestProjects/AppWithProjTool2Fx/App.csproj]: I think before we were just getting lucky that the projects were being built in the right order.

Comment (wli3) [reply TestAssets/DesktopTestProjects/AppWithProjTool2Fx/App.csproj]: @livarcocc I think during moving to arcade we make all test package to be in one place that is the "default"

File: test/Microsoft.DotNet.CommandFactory.Tests/GivenAProjectDependencyCommandResolver.cs [hunk 2]
```diff
@@ -30,8 +30,7 @@ public void ItReturnsACommandSpecWhenToolIsInAProjectRef()
             MSBuildTestProjectInstance =
                 TestAssets.Get("TestAppWithProjDepTool")
                     .CreateInstance()
-                    .WithSourceFiles()
-                    .WithRestoreFiles();
+                    .WithSourceFiles();
```
Comment (nguerrera) [inline test/Microsoft.DotNet.CommandFactory.Tests/GivenAProjectDependencyCommandResolver.cs]: Are there breaking changes in the new stage 0?

Comment (dsplaisted) [reply test/Microsoft.DotNet.CommandFactory.Tests/GivenAProjectDependencyCommandResolver.cs]: I don't think so, I'm still trying to figure out why some of the tests are failing with a missing packages directory.

**Comment by livarcocc** [comment]
I will take a look at these failures.

**Comment by wli3** [comment]
same failure here. https://github.com/dotnet/cli/pull/10818

**Pull request merged**